crates/testing/src/lib.rs

```diff
@@ -27,6 +27,14 @@ pub mod node_types;
 /// task to spin nodes up and down
 pub mod spinning_task;
 
+/// Implementations for testing/examples
+#[cfg_attr(
```

Comment:
Kinda hack-ish but can be useful IMO.
Not sure what's our general policy for this sort of thing though

Reply:
Is this `cfg_attr` needed because we are using demo implementations in non-test code? I think this is okay as a short-term fix but we'll need to investigate the relevant non-test code.

Do we have a list of places where demo implementations are used? If so we could probably investigate and fix them in a separate PR.

Reply:
It is not needed at all, it just generates warning if something from `demo` module is detected in context that is not one of test build, example, debug build

Reply:
Oh, I see. This is nice!

Reply:
I like the way you think. I don't fully understand what's achieved. We're already in the `testing` crate. So nothing from this crate should be used in non-test/example/debug contests. Can't we replace all these `cfg_attr` with a single one at the `testing` crate root?

Reply:
@ggutoski right, that makes sense, I'll see if gating entirety of the testing crate will cause any issues.